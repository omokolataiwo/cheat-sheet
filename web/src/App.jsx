import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Switch, Route, Redirect } from 'react-router-dom';
import { library } from '@fortawesome/fontawesome-svg-core';
import { faHeart, faBars } from '@fortawesome/free-solid-svg-icons';
import Home from './components/pages/Home';
import AuthUser from './components/pages/AuthUser';
import NavigationBar from './components/container/NavigationBar';
import { logout } from './actions/user';

library.add(faHeart, faBars);

export class App extends Component {
  state = {};

  render() {
    const {
      logout,
      history: { push }
    } = this.props;

    return (
      <div>
        <NavigationBar adminBar={!!localStorage.getItem('token')} />
        <Switch>
          <Route path="/home" component={Home} />
          <Route path="/user" component={AuthUser} />
          <Route
            path="/logout"
            component={() => {
              logout();
              push('/home');
              return null;
            }}
          />
          <Redirect to="/home" />
        </Switch>
      </div>
    );
  }
}

export default connect(
  ({ user }) => ({ user }),
  { logout }
)(App);
