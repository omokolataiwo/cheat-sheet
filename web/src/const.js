export const BASE_URL = process.env.NODE_ENV === 'development'
  ? 'http://localhost:7000/api/v1'
  : 'https://rc-git-sheet-cheat.herokuapp.com/api/v1';
