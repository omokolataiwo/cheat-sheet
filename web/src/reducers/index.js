import { combineReducers } from 'redux';
import cheat from './cheat';
import user from './user';

export default combineReducers({ cheat, user });
