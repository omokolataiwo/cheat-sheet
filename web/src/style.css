.cheat-masonry-grid {
  margin-top: 5%;
}
.category-section {
  border: 1px solid;
  border-color: #ccc;
  margin: 0.5%;
  padding: 1%;
  width: 30%;
}

.cheat {
  margin: 10px auto;
}

.cheat p {
  margin: 0;
}
.cheat .line {
  font-size: 12px;
  cursor: pointer;
}

.favorite{
  color: red;
  cursor: pointer;
}

.signin-form h3 {
  font-size: 1.7em;
  margin: 0;
}

.btn {
  background-color: #F05033;
}
.btn:hover {
  background-color: #1a237e;
}

.signin-form {
  width: 90%;
}
nav {
  background-color: #1a237e;
}
nav .brand-logo {
  margin-top: 6px;
  margin-left: 16px;
}
.sidenav {
  text-align: center;
  padding: 1em;
}

.sidenav-trigger {
  font-size: 2em;
}

.modal .modal-content {
  padding-bottom: 10%;
}