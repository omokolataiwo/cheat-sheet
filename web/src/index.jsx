import React from 'react';
import ReactDOM from 'react-dom';
import jwt from 'jsonwebtoken';
import { Provider } from 'react-redux';
import { BrowserRouter, Route } from 'react-router-dom';
import 'materialize-css/dist/css/materialize.min.css';
import 'materialize-css/dist/js/materialize';
import 'rc-tooltip/assets/bootstrap_white.css';
import App from './App';
import configureStore from './store';
import './style.css';
import { fetchAllCategoriesWithCheats } from './actions/cheat';
import { SIGNIN_SUCCESSFUL, INITIAL_USER_STATE } from './actions/type';

const store = configureStore();
const token = localStorage.getItem('token');

store.dispatch(fetchAllCategoriesWithCheats());

if (token) {
  store.dispatch({ type: SIGNIN_SUCCESSFUL, user: jwt.decode(token) });
  store.dispatch({ type: INITIAL_USER_STATE });
}

ReactDOM.render(
  <Provider store={store}>
    <BrowserRouter>
      <Route path="/" component={App} />
    </BrowserRouter>
  </Provider>,
  document.getElementById('app')
);

if (module.hot) {
  module.hot.accept();
}
