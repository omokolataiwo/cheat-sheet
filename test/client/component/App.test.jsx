import React from 'react';
import { shallow } from 'enzyme';
import { App } from '../../../web/src/App';

const props = {
  history: { push: jest.fn(() => {}) }
};

const wrapper = shallow(<App {...props} />);

describe('App Component', () => {
  it('should render component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
