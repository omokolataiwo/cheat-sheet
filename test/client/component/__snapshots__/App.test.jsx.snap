// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component should render component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    history={
      Object {
        "push": [MockFunction],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <_default
          adminBar={false}
        />,
        <Switch>
          <Route
            component={[Function]}
            path="/home"
          />
          <Route
            component={[Function]}
            path="/user"
          />
          <Route
            component={[Function]}
            path="/logout"
          />
          <Redirect
            push={false}
            to="/home"
          />
        </Switch>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "adminBar": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Route
              component={[Function]}
              path="/home"
            />,
            <Route
              component={[Function]}
              path="/user"
            />,
            <Route
              component={[Function]}
              path="/logout"
            />,
            <Redirect
              push={false}
              to="/home"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/home",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/user",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/logout",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "push": false,
              "to": "/home",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <_default
            adminBar={false}
          />,
          <Switch>
            <Route
              component={[Function]}
              path="/home"
            />
            <Route
              component={[Function]}
              path="/user"
            />
            <Route
              component={[Function]}
              path="/logout"
            />
            <Redirect
              push={false}
              to="/home"
            />
          </Switch>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "adminBar": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                component={[Function]}
                path="/home"
              />,
              <Route
                component={[Function]}
                path="/user"
              />,
              <Route
                component={[Function]}
                path="/logout"
              />,
              <Redirect
                push={false}
                to="/home"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/home",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/user",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/logout",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "push": false,
                "to": "/home",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
